free-3.md

```diff
@@ -10,41 +12,121 @@ _What need is met by your idea? Who are you helping? What is that person's
 experience like now? What would their experience be like if you could help 
 them?_
 
+A group is defined as a set of elements along with an operation such that the
+elements "play nicely" with the operation (the group is closed, there is an
+identity element, each element has an inverse, and the group is associative).
+Often times, it is useful to find correspondences between groups, known as
+homomorphisms, so that if a problem is complicated in one space, you can map it to
+another domain where the solution might be clearer.  Sometimes it is hard to come
+up with a map, and then to prove that the function is in fact a homomorphism, and
+to do so involves tediously checking the pairwise product of each element.  In
+abstract algebra, humans often tend to avoid doing brute force work in search of
+cleverer solutions. However, it would be helpful to have a computer check your
+progress on several examples to make sure you are on the right track.  This could
+be encoded in a DSL.  One could create a DSL where the user can specify groups and
+group operations and then leave the work to the DSL to find a bijection.
+Sometimes a computer wouldn't be able to determine a bijection this way, say if
+the group were infinite.  It would still be useful for a program to check hundreds
+of examples, and if all of them work then that is a good indication to the user
+that the groups are homomorphic.  In addition to showing two groups are
+homomorphic, there are many other useful facts that you can show with this DSL
+(_a_ is a subgroup of _b_, every group of size 5 or less is abelian) just by
+encoding the elements and having a computer try several possibilities.
```

Comment:
This doesn't seem like it really needs a DSL; just a normal program should be able to do it.